File: Data/qvs_sample_files/crypto_example.go
package main

import (
	"crypto/aes"
	_ "crypto/cipher"
	"crypto/rand"
	"crypto/rsa"
	"crypto/sha256"
	"fmt"
)

func main() {
	// RSA example
	privateKey, _ := rsa.GenerateKey(rand.Reader, 2048)

	// AES example
	key := make([]byte, 32)
	rand.Read(key)
	block, _ := aes.NewCipher(key)

	// SHA-256 example
	h := sha256.New()
	h.Write([]byte("hello world"))
	fmt.Printf("%x\n", h.Sum(nil))
}

File: Data/qvs_sample_files/crypto_example.js
const crypto = require('crypto');

// RSA example
const { publicKey, privateKey } = crypto.generateKeyPairSync('rsa', {
  modulusLength: 2048,
});

// AES example
const algorithm = 'aes-256-cbc';
const key = crypto.randomBytes(32);
const iv = crypto.randomBytes(16);

// SHA-256 example
const hash = crypto.createHash('sha256');
hash.update('Hello, World!');
console.log(hash.digest('hex'));
